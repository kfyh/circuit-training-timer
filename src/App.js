import React from 'react'
import PropTypes from 'prop-types'
import './App.css'
import CircuitTimer from './components/CircuitTimer'
import Selector from './components/Selector'

class App extends React.Component {
  constructor (props) {
    super(props)

    this.intervals = [
      { label: 'Burpies', time: 5, reps: 3, type: 'exercise' },
      {
        label: 'Finger Training',
        time: 10,
        reps: 7,
        type: 'set',
        steps: [
          { label: 'Hang for 7 secs', time: 60, reps: 1, type: 'exercise' },
          { label: 'Break and Recover', time: 240, reps: 1, type: 'break' }
        ]
      },
      { label: 'Finished', time: 5, reps: 1, type: 'prep' }
    ]
    this.state = {
      selectionScreen: true
    }
  }

  RenderScreen = (selectionScreen) => {
    let screen
    if (selectionScreen) {
      screen = <Selector onClick={this.onSelectionMade} />
    } else {
      screen = <CircuitTimer intervals={this.intervals} onComplete={this.onComplete}/>
    }
    return screen
  }

  render () {
    return (
      <div className="App">
        <div>Circuit Trainer Timer</div>
        {this.RenderScreen(this.state.selectionScreen)}
      </div>
    )
  }

  onSelectionMade = () => {
    this.setState({
      selectionScreen: false
    })
  }

  onComplete = () => {
    this.setState({
      selectionScreen: true
    })
  }
}

App.propTypes = {
  intervals: PropTypes.arrayOf(PropTypes.object)
}

export default App
