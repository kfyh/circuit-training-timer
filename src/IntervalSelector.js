export default function selectNextInterval (state, intervals) {
  let currentIntervalIndex = state.currentIntervalIndex
  let currentIntervalRep = state.currentIntervalRep
  let currentStepIndex = state.currentStepIndex
  let currentStepRep = state.currentStepRep
  const currentInterval = intervals[currentIntervalIndex]

  if (currentInterval.type === 'set') {
    const result = selectIntervalFromSet(state, intervals)
    currentIntervalIndex = result.currentIntervalIndex
    currentIntervalRep = result.currentIntervalRep
    currentStepIndex = result.currentStepIndex
    currentStepRep = result.currentStepRep
  } else if (currentIntervalRep < currentInterval.reps) {
    currentIntervalRep++
  } else {
    const nextIntervalIndex = currentIntervalIndex + 1
    if (nextIntervalIndex < intervals.length) {
      currentIntervalIndex = nextIntervalIndex
      currentIntervalRep = 1
    } else {
      currentIntervalIndex = -1
    }
  }

  if (currentIntervalIndex < 0) {
    return {
      currentIntervalIndex: -1,
      timeLeft: 0
    }
  } else {
    let newInterval = intervals[currentIntervalIndex]
    if (newInterval.type === 'set') {
      newInterval = newInterval.steps[currentStepIndex]
    }

    return {
      timeLeft: newInterval.time * 1000,
      endTime: Date.now() + (newInterval.time * 1000),
      currentIntervalIndex: currentIntervalIndex,
      currentIntervalRep: currentIntervalRep,
      currentStepIndex: currentStepIndex,
      currentStepRep: currentStepRep
    }
  }
}

function selectIntervalFromSet (state, intervals) {
  let currentIntervalIndex = state.currentIntervalIndex
  let currentIntervalRep = state.currentIntervalRep
  const currentInterval = intervals[currentIntervalIndex]

  let currentStepIndex = state.currentStepIndex
  let currentStepRep = state.currentStepRep
  const currentStep = currentInterval.steps[currentStepIndex]

  if (currentStepRep < currentStep.reps) {
    currentStepRep++
  } else {
    currentStepRep = 1
    const nextStepIndex = currentStepIndex + 1
    if (nextStepIndex < currentInterval.steps.length) {
      currentStepIndex = nextStepIndex
      currentStepRep = 1
    } else {
      if (currentIntervalRep < currentInterval.reps) {
        currentIntervalRep++
        currentStepIndex = 0
        currentStepRep = 1
      } else {
        const nextIntervalIndex = currentIntervalIndex + 1
        if (nextIntervalIndex < intervals.length) {
          currentIntervalIndex = nextIntervalIndex
          currentIntervalRep = 1
        } else {
          currentIntervalIndex = -1
        }
      }
    }
  }

  return {
    currentIntervalIndex: currentIntervalIndex,
    currentIntervalRep: currentIntervalRep,
    currentStepIndex: currentStepIndex,
    currentStepRep: currentStepRep
  }
}
