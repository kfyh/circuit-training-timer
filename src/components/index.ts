export * from './CircuitTimer';
export * from './Selector';
export * from './timer/index';
export * from './Dashboard';
