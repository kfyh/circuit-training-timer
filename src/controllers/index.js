export * from './FlattenedStepsController';
export * from './FlattenedCircuitController';