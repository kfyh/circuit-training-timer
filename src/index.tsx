import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { AppRouter } from './router/AppRouter';
import './index.css';
import { configureStore } from './store/configureStore';
import { addCircuit, addExercise, addExerciseGroup, selectCircuit } from './actions/selectorActions';

const store = configureStore();

store.dispatch(
	addExercise([
		{
			id: '0',
			name: 'Rest',
			description: 'Take a break',
		},
		{
			id: '1',
			name: '10 Push Ups',
			description: 'Up and Down',
		},
		{
			id: '2',
			name: '10 Sits Ups',
			description: 'Up and Down',
		},
		{
			id: '3',
			name: '10 Squats',
			description: 'Up and Down',
		},
		{
			id: '4',
			name: '10 Pull Ups',
			description: 'Up and Down',
		},
	])
);

store.dispatch(
	addExerciseGroup([
		{
			id: '0',
			name: 'Take a Break',
			exercises: [
				{
					exerciseId: '0',
					duration: 300,
					count: 1,
				},
			],
			repetitions: 1,
		},
		{
			id: '1',
			name: '50 Push Ups',
			exercises: [
				{
					exerciseId: '1',
					duration: 20,
					count: 10,
				},
				{
					exerciseId: '0',
					duration: 40,
					count: 1,
				},
			],
			repetitions: 5,
		},
		{
			id: '2',
			name: `Climber's Challenge`,
			exercises: [
				{
					exerciseId: '1',
					duration: 20,
					count: 10,
				},
				{
					exerciseId: '2',
					duration: 20,
					count: 10,
				},
				{
					exerciseId: '3',
					duration: 20,
					count: 10,
				},
				{
					exerciseId: '4',
					duration: 20,
					count: 10,
				},
				{
					exerciseId: '0',
					duration: 30,
					count: 1,
				},
			],
			repetitions: 10,
		},
	])
);

store.dispatch(
	addCircuit({
		id: '0',
		name: '50 Push Ups',
		exerciseGroups: [
			{
				id: '1',
				name: '50 Push Ups',
				exerciseGroupId: '1',
			},
		],
		repetitions: 1,
	})
);

store.dispatch(
	addCircuit({
		id: '1',
		name: `Climber's Challenge`,
		exerciseGroups: [
			{
				id: '1',
				name: '50 Push Ups',
				exerciseGroupId: '2',
			},
		],
		repetitions: 1,
	})
);

store.dispatch(
	selectCircuit({
		id: '0',
		name: '50 Push Ups',
		exerciseGroups: [
			{
				id: '1',
				name: '100 Push Ups, Sit Ups, Squats, Pull Ups',
				exerciseGroupId: '2',
			},
		],
		repetitions: 1,
	})
);

const rootElement = document.createElement('div');
rootElement.setAttribute('id', 'root');
document.body.appendChild(rootElement);
const jsx = (
	<Provider store={store}>
		<AppRouter />
	</Provider>
);

ReactDOM.render(jsx, rootElement);
