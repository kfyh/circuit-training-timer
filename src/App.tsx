import React, { ReactElement } from 'react';
import './App.css';
import { NavLink } from 'react-router-dom';

type AppState = {
	selectionScreen: boolean;
};

export class App extends React.Component<Record<string, unknown>, AppState> {
	constructor(props: Record<string, unknown>) {
		super(props);
		this.state = {
			selectionScreen: true,
		};
	}

	public render(): ReactElement {
		return (
			<div className="App">
				<div>Circuit Trainer Timer</div>
				<div>
					<NavLink to="/select" exact={true}>
						Select Training
					</NavLink>
					<br />
					<NavLink to="/timer" exact={true}>
						Start
					</NavLink>
				</div>
			</div>
		);
	}
}
