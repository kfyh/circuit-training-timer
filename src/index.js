import React from 'react'
import ReactDOM from 'react-dom'
import './index.css'
import App from './App'
import * as serviceWorker from './serviceWorker'

const intervals = [
  { label: 'Warmup', time: 5, reps: 1, type: 'prep' },
  { label: 'Porco push ups', time: 10, reps: 1, type: 'exercise' },
  { label: 'Break 1', time: 5, reps: 1, type: 'break' },
  {
    label: '3 Sets of Sit Ups',
    time: 10,
    reps: 3,
    type: 'set',
    steps: [
      { label: 'Porco sit ups yeah man', time: 8, reps: 1, type: 'exercise' },
      { label: 'Break 2', time: 3, reps: 1, type: 'break' }
    ]
  },
  { label: 'Cooldown', time: 5, reps: 1, type: 'prep' }
]

ReactDOM.render(
  <React.StrictMode>
    <App intervals={intervals}/>
  </React.StrictMode>,
  document.getElementById('root')
)

// If you want your app to work offline and load faster, you can change
// unregister() to register() below. Note this comes with some pitfalls.
// Learn more about service workers: https://bit.ly/CRA-PWA
serviceWorker.unregister()
