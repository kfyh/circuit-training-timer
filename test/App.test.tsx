import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../src/App';

test('When loading then title is Circuit Trainer Timer', () => {
	const render = shallow(<App />);
	expect(render).toMatchSnapshot();
});
