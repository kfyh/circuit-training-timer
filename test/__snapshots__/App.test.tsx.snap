// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When loading then title is Circuit Trainer Timer 1`] = `ShallowWrapper {}`;
